src/storage/src/compaction.rs

```diff
@@ -13,9 +13,22 @@
 // limitations under the License.
 
 mod dedup_deque;
+pub mod noop;
```

Comment:
Is it only used in tests? If so, we could add a `cfg(test)` attribute.

Reply:
mods marked with `#cfg(test)` cannot be shared between crates. These components are used in instance tests.